services/diversity/src/services/richness.js

```diff
@@ -12,105 +12,113 @@ module.exports = (RichnessPersistence, restAPI) => {
      *
      * @returns {Object[]} Number of inferred and observed species for the desired group.
      */
-    getNumberSpecies: async (areaType, areaId, group = 'all') => {
-      const data = [
-        {
-          id: 'total',
-          inferred: 40,
-          observed: 4,
-          region_observed: 110,
-          region_inferred: 130,
-        },
-        {
-          id: 'endemic',
-          inferred: 20,
-          observed: 25,
-          region_observed: 120,
-          region_inferred: 100,
-        },
-        {
-          id: 'invasive',
-          inferred: 20,
-          observed: 10,
-          region_observed: 90,
-          region_inferred: 80,
-        },
-        {
-          id: 'threatened',
-          inferred: 15,
-          observed: 20,
-          region_observed: 90,
-          region_inferred: 100,
-        },
-      ];
+    getNumberOfSpecies: async (areaType, areaId, group = 'all') => {
+      const promises = [];
       switch (group) {
         case 'total':
-          return [data[0]];
+          promises.unshift(RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId));
+          break;
         case 'endemic':
-          return [data[1]];
+          promises.unshift(RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId));
+          break;
         case 'invasive':
-          return [data[2]];
+          promises.unshift(RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId));
+          break;
         case 'threatened':
-          return [data[3]];
+          promises.unshift(RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId));
+          break;
         case 'all':
+          promises.unshift(
+            RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId),
+          );
+          break;
         default:
-          return data;
+          return [];
       }
+      return Promise.all(promises)
+        .then((response) => {
+          if (group !== 'all') {
+            return response.map((item) => {
+              if (item.length > 0) return { id: group, ...item[0] };
+              throw new Error('Error no data is found');
```

Comment:
I don't remember if we agreed on something, but if there are no data, should we return an error? I thought it would return an empty array.

Reply:
Yes, we did set item as return then I changed because in the case of empty values it does not return any error when empty values are returned. Just when I set like this.  Last commit leave the returns as we initially agreed.